docs/howto/prepare-and-run.md

```diff
@@ -0,0 +1,70 @@
+(prepare-and-run)=
+# How to prepare and run a SQLair query
+
+A SQLair query is just like a regular SQL query, except that it contains special
+input and output expressions. These indicate to SQLair the Go objects that your
+inputs and outputs correspond to. To create and run a SQLair query there are
+several steps:
+1. Ensure the types you want to map to the query inputs and outputs are in the
+   right form for SQLair.
+2. Write your query with SQLair syntax.
+3. Prepare your query with `sqlair.Prepare`.
+4. Run your query on a database or transaction.
+
+## Prepare your Go types
+
+See {ref}`types` for how to prepare your types.
+
+## Write your query
+
+SQLair queries are regular SQL queries with special input and output expressions
+inserted which allow you to reference the Go types directly from the SQL. You
+can have any number of input and output expressions in your query.
+
+To write expressions that map Go types to query inputs see {ref}`input-expressions`.
+
+To write expressions that map Go types to query outputs see {ref}`output-expressions`.
+
+If a query contains no output expressions then SQLair will not fetch any results
+from the database. This means that you will not get a `ErrNoRows` error if you
+run a `SELECT` statement with no output expressions to check if a row exists.
+You will need to include a placeholder output expression.
+
+## Prepare your query
+
+Now you have your query string, you need to pass it to `sqlair.Prepare` along
+with all the types mentioned in the input and output expression.
+
+See more [`sqlair.Preapre`](https://pkg.go.dev/github.com/canonical/sqlair#Prepare)
+
+This is needed so that SQLair can gather reflection information about the types
+and verify the input/output expressions. If there is an issue with the actual
+SQL in the query this will not catch it.
+
+Note that this will not prepare the query on the database. In fact, the
+`sqlair.Statement` you get back here can be used on any database or transaction.
+The query is automatically prepared against the database the first time it is
+run. As a user, you do not need to worry about managing the driver prepared
+statements on the database. SQLair manages this for you.
+
```

Comment:
This is a lot of words. Wondering how to tell the same story more efficiently.

Reply:
One way to do that would be not to tell it here, this is not strictly relevant.

Reply:
I think I'll leave this here for now, but would like to find a new home for it in the future. There a bit more that can be said on the subject so I think putting it in here doesn't do it justice. Its quite a cool feature of SQLair